pkg/addons/vpc_controller.go

```diff
@@ -209,9 +212,52 @@ func (v *VPCController) createCertSecrets(key, cert []byte) error {
 }
 
 func (v *VPCController) deployVPCResourceController() error {
+	irsaEnabled, err := v.irsa.IsSupported()
+	if err != nil {
+		return err
+	}
+	if irsaEnabled {
+		sa := &api.ClusterIAMServiceAccount{
+			ClusterIAMMeta: api.ClusterIAMMeta{
+				Name:      vpcControllerName,
+				Namespace: vpcControllerNamespace,
+			},
+			AttachPolicy: map[string]interface{}{
```

Comment:
could this be its own var/function like in `pkg/cfn/builder`?

Reply:
Makes sense :+1: